app/scripts/modules/core/src/pipeline/config/templates/v2/configurePipelineTemplateModalV2.controller.ts

```diff
@@ -0,0 +1,251 @@
+import { module, IController, IScope, IHttpPromiseCallbackArg, IPromise } from 'angular';
+import { IModalInstanceService } from 'angular-ui-bootstrap';
+import { without, chain, has } from 'lodash';
+
+import { Application } from 'core/application/application.model';
+import { IPipelineTemplateConfigV2 } from 'core/domain';
+import { PipelineTemplateV2Service } from 'core/pipeline';
+import { VariableValidatorService } from '../validators/variableValidator.service';
+
+import {
+  PipelineTemplateReader,
+  IVariableMetadata,
+  IPipelineTemplatePlanResponse,
+  IPipelineTemplate,
+  IPipelineTemplatePlanError,
+} from 'core/pipeline/config/templates/PipelineTemplateReader';
+import { IVariable } from '../inputs/variableInput.service';
+
+export interface IVariableMetadataGroup {
+  name: string;
+  variableMetadata: IVariableMetadata[];
+}
+
+export interface IState {
+  error: boolean;
+  loading: boolean;
+  loadingError: boolean;
+  noVariables: boolean;
+  planErrors: IPipelineTemplatePlanError[];
+  inheritTemplateParameters: boolean;
+  inheritTemplateTriggers: boolean;
+  inheritTemplateNotifications: boolean;
+}
+// Logic lifted from ConfigurePipelineTemplateModalController and slightly refactored to support MPTV2.
+// TODO: Once MPTV1 is fully migrated, delete ConfigurePipelineTemplateModalController.
+export class ConfigurePipelineTemplateModalV2Controller implements IController {
+  public pipelineName: string;
+  public variableMetadataGroups: IVariableMetadataGroup[];
+  public variables: IVariable[];
+  public state: IState = {
+    loading: true,
+    error: false,
+    planErrors: null,
+    loadingError: false,
+    noVariables: false,
+    inheritTemplateParameters: true,
+    inheritTemplateTriggers: true,
+    inheritTemplateNotifications: true,
+  };
+  private template: IPipelineTemplate;
+  private source: string;
+
+  public static $inject = [
+    '$scope',
+    '$uibModalInstance',
+    'application',
+    'pipelineTemplateConfig',
+    'isNew',
+    'pipelineId',
+    'executionId',
+  ];
+  constructor(
+    private $scope: IScope,
+    private $uibModalInstance: IModalInstanceService,
+    private application: Application,
+    public pipelineTemplateConfig: IPipelineTemplateConfigV2,
+    public isNew: boolean,
+    private pipelineId: string,
+    private executionId: string,
+  ) {}
+
+  public $onInit(): void {
+    this.initialize();
+  }
+
+  public initialize(): void {
+    const { exclude, name, template } = this.pipelineTemplateConfig;
+
+    const excluded = exclude || [];
```

Comment:
triggers, notifications, and parameters are now automatically inherited from templates unless the pipeline opts out by defining an exclude array